pkg/providers/vsphere/virtualmachine/backup.go

```diff
@@ -55,22 +58,30 @@ func BackupVirtualMachine(opts BackupVirtualMachineOptions) error {
 		opts.VMCtx.Logger.Error(err, "failed to get VM properties for backup")
 		return err
 	}
-
 	curExCfg := pkgutil.OptionValues(moVM.Config.ExtraConfig)
 	var ecToUpdate pkgutil.OptionValues
 
-	vmYAML, err := getDesiredVMResourceYAMLForBackup(opts.VMCtx.VM, curExCfg)
-	if err != nil {
-		opts.VMCtx.Logger.Error(err, "failed to get VM resource yaml for backup")
-		return err
-	}
-	if vmYAML == "" {
-		opts.VMCtx.Logger.V(4).Info("Skipping VM resource yaml backup as unchanged")
-	} else {
-		ecToUpdate = append(ecToUpdate, &vimtypes.OptionValue{
-			Key:   vmopv1.VMResourceYAMLExtraConfigKey,
-			Value: vmYAML,
-		})
+	/*
+	 * When VMIncrementalRestore FSS is enabled, perform backups when
+	 * - no backup version annotation is present.
+	 * - the backup version annotation matches the current backup version in extraConfig.
+	 * - the backup version annotation is older than the extraConfig backup version key. This can only happen when reconfigure succeeds to add a new
+	 *   extraConfig backup version but the next reconcile failed to update the annotation on the VM resource.
+	 */
+	if pkgcfg.FromContext(opts.VMCtx).Features.VMIncrementalRestore {
```

Comment:
// I am worried that this might race with us persisting the backup a few lines below.  Think what happens if the VM is restored after this check succeeds (isBackupNeeded returns true).  In that case, the backup stored in the VM will be replaced a few lines below before registration has had a chance to run.  IMHO, this check should be as close to us trying to persist the backup as possible.

Reply:
makes sense. 
- We'll also be prepping a backup if we don't check early on too in the method. So we might need to check early on to prevent extra work being done.
- Also to reduce chances for the race as you mentioned, the properties need to be fetched again from the moVM right before trying Reconfigure() for backup. The extraConfig properties values fetched earlier in the method is what gets used until the end. So unless we fetch again, this race will still happen.

Reply:
I don't think vendors treat VM Service VMs differently, and it's kind of impractical to have all vendors adopt this approach. Technically, VI Admins can pause the VM reconciliation before performing restore but it won't be an automated backup & restore workflow :)

Reply:
Arunesh and I spoke about this offline. I've added to a todo to handle this minor case where a race could happen. resolving this for now